Fix tests to support parallel execution with pytest-xdist (#593)
aio-libs/aiosmtpd: aiosmtpd/docs/controller.rst

    :param hostname: Will be given to the event loop's :meth:`~asyncio.loop.create_server` method
       as the ``host`` parameter, with a slight processing (see below)
    :type hostname: Optional[str]
    :param port: Will be passed-through to :meth:`~asyncio.loop.create_server` method.

>         If ``0``, a random unused (ephemeral) port will be selected.

Reviewer: We probably should also have a word of warning here too (untested, probably needs a syntax check, adjustments)
```suggestion
        If ``0``, the OS kernel will select a random unused (ephemeral)
        port when the server starts listening to the socket.

        .. caution::

           You can access the :attr:`.requested_port` at any time but
           attempting to resolve the :attr:`.port` prior to binding
           to the socket will error out.
```

*(https://aiosmtpd--593.org.readthedocs.build/en/593/controller.html#aiosmtpd.controller.Controller)*

Author: Thanks, good call, updated.

    :param hostname: Will be given to the event loop's :meth:`~asyncio.loop.create_server` method
       as the ``host`` parameter, with a slight processing (see below)
    :type hostname: Optional[str]
    :param port: Will be passed-through to :meth:`~asyncio.loop.create_server` method.

>         If ``0``, the OS kernel will select a random unused (ephemeral) port

Reviewer: We probably should also have a word of warning here too (untested, probably needs a syntax check, adjustments)
```suggestion
        If ``0``, the OS kernel will select a random unused (ephemeral)
        port when the server starts listening to the socket.

        .. caution::

           You can access the :attr:`.requested_port` at any time but
           attempting to resolve the :attr:`.port` prior to binding
           to the socket will error out.
```

*(https://aiosmtpd--593.org.readthedocs.build/en/593/controller.html#aiosmtpd.controller.Controller)*

Reviewer: (this was sitting in my browser unsent from the previous pass, for some reason)

Author: But isn't this the description of constructor arguments? The instance attributes are documented further below.

    In addition to those provided by :class:`BaseController`,
    this class provides the following:

    .. attribute:: hostname
        :type: str

        The *hostname* of the socket the server is listening on.

        If the server is not running (because it has not been started yet, or has
        already been stopped), accessing this attribute will raise

>         a :class:`RuntimeError`.

Reviewer: ```suggestion
        a :exc:`RuntimeError`.
```

        If an empty hostname parameter was passed to the controller's constuctor
        then the server is running on all available interfaces, that may have
        different hostnames. This property returns the hostname of the first listening
        socket, but the order of listening sockets is not defined.

        To access the hostname parameter that was passed in to this controller's
        constructor, use :attr:`requested_hostname`.

    .. attribute:: port
        :type: int

        The *port* of the socket the server is listening on.

        If the server is not running (because it has not been started yet, or has
        already been stopped), accessing this attribute will raise

>         a :class:`RuntimeError`.

Reviewer: exceptions have a dedicated role in RST
```suggestion
        a :exc:`RuntimeError`.
```

Author: Thanks, fixed.
